src/App.tsx

```diff
@@ -1,27 +1,94 @@
-import React from 'react';
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import { Oval } from 'react-loader-spinner';
+import 'bulma';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import React, { useCallback, useState } from 'react';
+import { GoodsList } from './components/GoodsList';
+import { Button } from './types/Button';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+const buttons: Button[] = [
+  {
+    text: 'Load all goods',
+    loadFunction: getAll,
+    dataCy: 'all-button',
+  },
+  {
+    text: 'Load 5 first goods',
+    loadFunction: get5First,
+    dataCy: 'first-five-button',
+  },
+  {
+    text: 'Load red goods',
+    loadFunction: getRedGoods,
+    dataCy: 'red-button',
+  },
+];
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const loadGoods = useCallback(async (loadFunction: () => Promise<Good[]>) => {
```

Comment:
Would be nice to move this function type to separate type, but it's not critical